src/Aspire.Hosting.Azure.ServiceBus/AzureServiceBusExtensions.cs

```diff
@@ -124,19 +151,381 @@ public static IResourceBuilder<AzureServiceBusResource> AddQueue(this IResourceB
+            }
+        });
+
         return builder;
     }
+
+    /// <summary>
+    /// Configures an Azure Service Bus resource to be emulated. This resource requires an <see cref="AzureServiceBusResource"/> to be added to the application model.
+    /// </summary>
+    /// <remarks>
+    /// This version of the package defaults to the <inheritdoc cref="ServiceBusEmulatorContainerImageTags.Tag"/> tag of the <inheritdoc cref="ServiceBusEmulatorContainerImageTags.Registry"/>/<inheritdoc cref="ServiceBusEmulatorContainerImageTags.Image"/> container image.
+    /// </remarks>
+    /// <param name="builder">The Azure Service Bus resource builder.</param>
+    /// <param name="configureContainer">Callback that exposes underlying container used for emulation to allow for customization.</param>
+    /// <returns>A reference to the <see cref="IResourceBuilder{T}"/>.</returns>
+    /// <example>
+    /// The following example creates an Azure Service Bus resource that runs locally is an emulator and referencing that
+    /// resource in a .NET project.
+    /// <code lang="csharp">
+    /// var builder = DistributedApplication.CreateBuilder(args);
+    ///
+    /// var serviceBus = builder.AddAzureServiceBus("myservicebus")
+    ///    .RunAsEmulator()
+    ///    .AddQueue("queue");
+    ///
+    /// builder.AddProject&lt;Projects.InventoryService&gt;()
+    ///        .WithReference(serviceBus);
+    ///
+    /// builder.Build().Run();
+    /// </code>
+    /// </example>
+    public static IResourceBuilder<AzureServiceBusResource> RunAsEmulator(this IResourceBuilder<AzureServiceBusResource> builder, Action<IResourceBuilder<AzureServiceBusEmulatorResource>>? configureContainer = null)
+    {
+        if (builder.ApplicationBuilder.ExecutionContext.IsPublishMode)
+        {
+            return builder;
+        }
+
+        // Add emulator container
+        var configHostFile = Path.Combine(Path.GetTempPath(), "AspireServiceBusEmulator", Path.GetRandomFileName() + ".json");
+
+        Directory.CreateDirectory(Path.GetDirectoryName(configHostFile)!);
+
+        var password = PasswordGenerator.Generate(16, true, true, true, true, 0, 0, 0, 0);
+
+        var customMountAnnotation = new ContainerMountAnnotation(
+                configHostFile,
+                AzureServiceBusEmulatorResource.EmulatorConfigJsonPath,
+                ContainerMountType.BindMount,
+                isReadOnly: false);
+
+        builder
+            .WithEndpoint(name: "emulator", targetPort: 5672)
+            .WithAnnotation(new ContainerImageAnnotation
+            {
+                Registry = ServiceBusEmulatorContainerImageTags.Registry,
+                Image = ServiceBusEmulatorContainerImageTags.Image,
+                Tag = ServiceBusEmulatorContainerImageTags.Tag
+            })
+            .WithAnnotation(customMountAnnotation);
+
+        var sqlEdgeResource = builder.ApplicationBuilder
+                .AddContainer($"{builder.Resource.Name}-sqledge",
```

Comment:
Do we have to use sqledge? Or can/should we reuse our own Sql hosting package?

Reply:
Good question, will ask the team. I already asked about the warning that sqledge was getting deprecated but did not get an answer AFAIK.

Reply:
It should work with SQL Server, a differentiating factor is that sqledge supports ARM, and hence MacOS. It also uses less resources, but not sure it's a deciding factor.